File: internal/model/site_link.go
package model

// SiteLink 常用网站（友情链接）模型
type SiteLink struct {
	BaseModel
	Name        string `gorm:"column:name;type:varchar(50);not null" json:"name"`             // 网站名称
	URL         string `gorm:"column:url;type:varchar(255);not null" json:"url"`              // 链接地址
	Icon        string `gorm:"column:icon;type:varchar(500);not null;default:''" json:"icon"` // 图标图片 URL
	Description string `gorm:"column:description;type:varchar(200)" json:"description"`       // 可选描述
	Sort        int    `gorm:"column:sort;not null;default:0" json:"sort"`                    // 排序，小靠前
	Status      int16  `gorm:"column:status;not null;default:1" json:"status"`                // 1 启用 0 停用
}

// TableName 指定表名
func (SiteLink) TableName() string {
	return "blog_link"
}

File: web/src/api/siteLink.ts
import request from './request';
import type { ApiResponse } from './request';
import type { SiteLink } from './types';

/** 新增/更新常用网站的请求体 */
export interface SiteLinkSaveParams {
  name: string;
  url: string;
  /** 图标图片 URL */
  icon: string;
  description: string;
  sort: number;
  /** 1 启用 0 停用 */
  status: number;
}

/** 获取全部常用网站（含停用，按 sort 升序） */
export function getSiteLinks() {
  return request.get<unknown, ApiResponse<SiteLink[]>>('/site/links');
}

/** 新增常用网站 */
export function createSiteLink(data: SiteLinkSaveParams) {
  return request.post<unknown, ApiResponse<null>>('/site/links', data);
}

/** 更新常用网站 */
export function updateSiteLink(id: number, data: SiteLinkSaveParams) {
  return request.put<unknown, ApiResponse<null>>(`/site/links/${id}`, data);
}

/** 删除常用网站 */
export function deleteSiteLink(id: number) {
  return request.delete<unknown, ApiResponse<null>>(`/site/links/${id}`);
}
